sql/driver/driver_test.go

```diff
@@ -149,6 +149,9 @@ func TestCreateTable(t *testing.T) {
 	if _, err := db.Exec("CREATE TABLE IF NOT EXISTS users " + cols); err != nil {
 		t.Fatal(err)
 	}
+	if _, err := db.Exec("CREATE TABLE employees " + cols); err != nil {
```

Comment:
Yes, the session state is local to the connection. That is the desired semantics. Something like a `USE` statement is almost never performed directly by an application, but by the driver itself when it initializes the connection. And transaction session state should be tied to the connection.